cloud/ec2_fleet.go

```diff
@@ -475,7 +514,13 @@ func (m *ec2FleetManager) makeOverrides(ctx context.Context, ec2Settings *EC2Pro
 	if len(subnets) > 0 {
 		overrides := make([]*ec2.FleetLaunchTemplateOverridesRequest, 0, len(subnets))
 		for _, subnet := range subnets {
-			overrides = append(overrides, &ec2.FleetLaunchTemplateOverridesRequest{SubnetId: aws.String(subnet.SubnetID)})
+			supported, err := typeCache.azSupportsInstanceType(ctx, m.client, subnet.AZ, ec2Settings.InstanceType)
+			if err != nil {
+				return nil, errors.Wrapf(err, "can't get supported instance types for AZ '%s'", subnet.AZ)
+			}
+			if supported {
+				overrides = append(overrides, &ec2.FleetLaunchTemplateOverridesRequest{SubnetId: aws.String(subnet.SubnetID)})
+			}
 		}
 
 		return overrides, nil
```

Comment:
Are we okay returning an empty slice if there are no supported subnets?

Reply:
that's a good question.
As written we return an empty slice and the instance will be requested with the default subnet set on the distro.